backend/controller/SpotifyController.js

```diff
@@ -123,6 +123,98 @@ exports.getMyRecentlyPlayedTracks = (spotifyId, options) => {
     });
 }
 
+/**
+ * Gets recomendations based on most recently played tracks
+ * @param {*} spotifyId 
+ * @returns list of tracks to recommend user
+ */
+exports.getRecommendations = async (spotifyId) => {
+  const client = spotifyClients[spotifyId];
+
+  // Get most recently played songs
+  const recentlyPlayedData = await this.getMyRecentlyPlayedTracks(spotifyId, { limit: 5 });
+  console.log('data', recentlyPlayedData)
+  const recentlyPlayedSongs = recentlyPlayedData.body.items;
+
+  // Randomly determine amount of songs to use as seed and amount of artists to use as seed
+  // The combination should equal to 5
+  const amountOfSongs = Math.floor(Math.random() * 4) + 1;
+  const amountOfArtists = 5 - amountOfSongs;
+  const artistSeed = [];
+  const songSeed = [];
+
+  // Go through songs and add them to artistSeed and songSeed if the lists aren't full
+  for (const song of recentlyPlayedSongs) {
+    if (artistSeed.length < amountOfArtists) {
+      artistSeed.push(song.track.artists[0].id);
+    }
+    if (songSeed.length < amountOfSongs) {
+      songSeed.push(song.track.id);
+    }
+    if (songSeed.length >= amountOfSongs && artistSeed.length >= amountOfArtists) {
+      break;
+    }
+  }
+
+  console.log('song seed', songSeed, 'artist seed', artistSeed);
+
+  return client.getRecommendations({
+    seed_songs: songSeed,
+    seed_artists: artistSeed,
+    min_popularity: 20,
+    limit: 10
+  }).then(
+    (data) => { return data; },
+    (_err) => {
+      console.log('Something went wrong!', err);
+      throw err;
+    });
+}
+
+/**
+ * Gets recomendations based on most recently played tracks
```

Comment:
yoink

Reply:
I got a surprise for you 🤭

Test it though, It works on my end but I'm never done code like this LOL